core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1267,21 +1444,207 @@ class KafkaApis(val requestChannel: RequestChannel,
       )
 
       // call the share partition manager to fetch messages from the local replica
-      sharePartitionManager.fetchMessages(
-        groupId,
-        shareFetchRequest.data.memberId,
-        params,
-        interesting.asJava
-      ).whenComplete { (responsePartitionData, throwable) =>
-        if (throwable != null) {
+      try{
+        val responsePartitionData = sharePartitionManager.fetchMessages(
+          groupId,
+          shareFetchRequest.data.memberId,
+          params,
+          interesting.asJava
+        ).get()
+        processResponseCallback(responsePartitionData.asScala.toMap)
+      } catch {
+        case e: Throwable => throw e
+      }
+    }
+  }
+
+  /**
+   * Handle a shareFetch request
+   */
+  def handleShareFetchRequest(request: RequestChannel.Request): Unit = {
+    val shareFetchRequest = request.body[ShareFetchRequest]
+
+    if (!config.isShareGroupEnabled) {
+      // The API is not supported when the configuration `group.share.enable` has not been set
+      requestHelper.sendMaybeThrottle(request, shareFetchRequest.getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
+      CompletableFuture.completedFuture[Unit](())
+      return
+    }
+    val topicNames = metadataCache.topicIdsToNames()
+    val sharePartitionManager : SharePartitionManager = sharePartitionManagerOption match {
+      case Some(manager) => manager
+      case None => throw new IllegalStateException("ShareFetchRequest received but SharePartitionManager is not initialized")
+    }
+
+    val groupId = shareFetchRequest.data.groupId
+    val clientId = request.header.clientId
+    val memberId = shareFetchRequest.data().memberId()
+    val shareSessionEpoch = shareFetchRequest.data().shareSessionEpoch()
+
+    val shareFetchData = shareFetchRequest.shareFetchData(topicNames)
+    val forgottenTopics = shareFetchRequest.forgottenTopics(topicNames)
+
+    // TODO : replace this initialization when share fetch session metadata is sent by the client in the shareFetchRequest
+    val newReqMetadata : ShareFetchMetadata = new ShareFetchMetadata(Uuid.fromString(memberId), shareSessionEpoch)
```

Comment:
I noticed there are integration test failures with respect to Share Fetch. I debugged it locally and saw that the `memberId` getting populated inside `handleShareFetchRequest()` is coming out as empty string which could be causing the issue in this operation `Uuid.fromString(memberId)`

Reply:
This has been fixed in the client I think. Rebasing on the kip-932 branch should resolve it.